modules/samples/oauth2/playground2/src/main/org/wso2/sample/identity/oauth2/IDTokenDecrypterServlet.java

```diff
@@ -0,0 +1,121 @@
+/*
+ *Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *WSO2 Inc. licenses this file to you under the Apache License,
+ *Version 2.0 (the "License"); you may not use this file except
+ *in compliance with the License.
+ *You may obtain a copy of the License at
+ *
+ *http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *Unless required by applicable law or agreed to in writing,
+ *software distributed under the License is distributed on an
+ *"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *KIND, either express or implied.  See the License for the
+ *specific language governing permissions and limitations
+ *under the License.
+ */
+
+package org.wso2.sample.identity.oauth2;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.crypto.RSADecrypter;
+import com.nimbusds.jwt.EncryptedJWT;
+import net.minidev.json.JSONObject;
+import java.io.IOException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.text.ParseException;
+import java.util.Base64;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet class for decrypting id tokens
+ */
+@WebServlet(name = "IDTokenDecrypterServlet")
+public class IDTokenDecrypterServlet extends HttpServlet {
+
+    private static Logger LOGGER = Logger.getLogger(ContextEventListener.class.getName());
+
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String idToken = request.getParameter("idToken");
+        String privateKeyString = request.getParameter("privateKeyString");
+
+        ServletOutputStream out = response.getOutputStream();
+
+        if (idToken.equals("") || privateKeyString.equals("")) {
```

Comment:
When the id_token is empty this will return "Client private key cannot be empty!".
Better to do separate checks or change the error message to say "Client private key and id_token cannot be empty!"

Reply:
I'll do separate checks.